example/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+cmake_minimum_required(VERSION 3.13)
+
+project(example)
+
+set(SOURCES
+  example.cpp
+)
+
+add_executable(example.bin ${SOURCES})
+
+target_link_libraries(example.bin ads pthread)
+
+target_compile_features(example.bin PRIVATE
+  cxx_std_11
+)
+
+target_compile_options(example.bin PRIVATE
```

Comment:
I am not a CMake expert, could we put this in the toplevel CMakeList? As all our subdirs _should_ use the same compile features and options.